hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

```diff
@@ -278,5 +299,16 @@ public HoodieIndexConfig build() {
       HoodieIndex.IndexType.valueOf(props.getProperty(INDEX_TYPE_PROP));
       return config;
     }
+
+    private String getDefaultIndexType(EngineType engineType) {
+      switch (engineType) {
```

Comment:
Are there other configurations that are different in different engines, and if there are many, are they easy to implement in this way?

Reply:
> Are there other configurations that are different in different engines, and if there are many, are they easy to implement in this way?

Currently, I recognized this only as I am working on flink index. 
I think it is simple, just add a check step before initialized it.
this way should be ok.